sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -120,7 +120,13 @@ object JDBCRDD extends Logging {
       case StringStartsWith(attr, value) => s"${attr} LIKE '${value}%'"
       case StringEndsWith(attr, value) => s"${attr} LIKE '%${value}'"
       case StringContains(attr, value) => s"${attr} LIKE '%${value}%'"
-      case In(attr, value) => s"$attr IN (${compileValue(value)})"
+      case In(attr, value) =>
+        if (value.nonEmpty) {
+          s"$attr IN (${compileValue(value)})"
+        } else {
+          // Return false literal when value is empty.
```

Comment:
This is dead code, right? 

If Optimizer already replace the empty value by a constant `false`, how to reach this branch?